KenticoOnboardingApplication/src/KenticoOnboardingApplication.Contracts/Services/Wrappers/RetrievedItem.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System;
+
+namespace KenticoOnboardingApplication.Contracts.Services.Wrappers
+{
+    public class RetrievedItem<TItem> where TItem : class
+    {
+        public TItem Item { get; }
+        public bool WasFound => Item != null;
+        public static RetrievedItem<TItem> Empty = new RetrievedItem<TItem>();
```

Comment:
Love this simplicity and readability this instance concept brings 😍 
```suggestion
        public static RetrievedItem<TItem> Empty { get; } = new RetrievedItem<TItem>();
```
(the change is subtle, however, properties (in general) bring much more powerful options to omit breaking changes in future)

JFYI: As an alternative I don't want You to implement, one can introduce a static construction method (e.g. `RetrievedItem.Create<TItem>(TItem item) => item == null ? Empty : new GenericRetrievedItem<TItem>(item) `) - for example, you can see this other approach in [this other repository](https://github.com/petrareb/kentico-onboarding-cs/blob/d0199b126355426185e977fc8270500652a1c312/MyOnboarding/src/MyOnboardingApp.Contracts/Validation/ResolvedItem.cs))